internal/compiler/output_columns.go

```diff
@@ -713,3 +718,17 @@ func findColumnForRef(ref *ast.ColumnRef, tables []*Table, targetList *ast.List)
 
 	return nil
 }
+
+// hasWhereOrHavingClause returns true if the statement contains WHERE or HAVING clause
+func hasWhereOrHavingClause(node ast.Node) bool {
+	stmt := node.(*ast.SelectStmt)
+
+	if _, isTODO := stmt.WhereClause.(*ast.TODO); stmt.WhereClause != nil && !isTODO {
+		return true
+	}
+	if _, isTODO := stmt.HavingClause.(*ast.TODO); stmt.HavingClause != nil && !isTODO {
+		return true
+	}
+
+	return false
+}
```

Comment:
I found my mistake now.

I thought that It's ok to generate null if a subselect statement has where or having clauses.
However, It's not enough. In case of no recored, the subselect is able to be null without where or having causes.